Terminal.Gui/Core/View.cs

```diff
@@ -989,7 +989,7 @@ public override bool ProcessHotKey (KeyEvent keyEvent)
 			if (subviews == null || subviews.Count == 0)
 				return false;
 			foreach (var view in subviews)
-				if (view.ProcessHotKey (keyEvent))
+				if (view.SuperView.IsCurrentTop && view.ProcessHotKey (keyEvent))
```

Comment:
This comment is broader than this PR, but anytime I see three functions that have EXACTLY the same logic in them, with a fair amount of logic, I wonder if they could be refactored to share implementation to reduce code and risk of one of them having a bug.

ARE all 3 the same???

Reply:
These processes have different behaviours but with the same logic. They were already there and I don't wanna break anything. Later we can see if it will be possible to congregate all of them.